utility/src/main/java/org/oppia/util/data/HtmlParser.kt

```diff
@@ -0,0 +1,38 @@
+package org.oppia.util.data
+
+import android.content.Context
+import android.text.Html
+import android.text.Spannable
+import android.widget.TextView
+
+private const val CUSTOM_TAG = "oppia-noninteractive-image"
+private const val HTML_TAG = "img"
+private const val CUSTOM_ATTRIBUTE = "filepath-with-value"
+private const val HTML_ATTRIBUTE = "src"
+
+/** Html Parser for android TextView to parse Html tag. */
+class HtmlParser(internal var context: Context) {
+
+  /***
+   * This method is called to parse HTML from a string-content, it manipulates the HTML elements, attributes and text
```

Comment:
It might be nice to indicate that this is for replacing custom Oppia tags. Maybe something like:

```kotlin
/**
 * Replaces custom Oppia tags with Android-compatible versions for a given raw HTML string, and returns the HTML [Spannable].
 * ...
```

Reply:
Updated comment.